http/src/main/java/io/micronaut/http/body/DefaultMessageBodyHandlerRegistry.java

```diff
@@ -150,10 +153,10 @@ protected <T> MessageBodyWriter<T> findWriterImpl(Argument<T> type, List<MediaTy
             if (exactMatch.size() == 1) {
                 return beanLocator.getBean(exactMatch.iterator().next());
             } else {
-                // pick highest priority
-                return OrderUtil.sort(beanDefinitions.stream())
-                    .findFirst()
+                // Pick the highest priority
+                return beanDefinitions.stream()
```

Comment:
is this a functional change? it seems like this flips the order

Reply:
It picks the highest priority as the comment suggests it should. I don't know what the correct behavior should be, so it can be flipped if needed.

Reply:
it should comply with Ordered.HIGHEST_PRECEDENCE and LOWEST_PRECEDENCE, so it should be flipped